### DISCUSSION TITLE
Question/suggestion: units training data

### DISCUSSION THREAD

**Discussion by mm12** [LAION-AI/Open-Assistant#3447] [Ideas]
When doing math, there should be an integration of unit conversion, similar to when using the calculator in Google's omnibox (for example, type `(500mA) * (1Ω)` into the chrome url bar, it converts to volts). #595 would partially do this, but I have found wolfram often doesn't understand what is a variable and a unit, and what unit it is (ie, instead of Amperes, it guesses acres). I'm not sure if other Wolfram products (like compute or mathematica) would solve this, and even Google's calculator has faults (doesn't like imaginary numbers, kiloOhms (kΩ), µA, etc, unless you type it out). 

One way this could be solved is by trying to extract any units and computing them separately.

**Comment by andreaskoepf** [comment]
What you suggest touches a more fundamental problem: How to properly add math capabilities to a LLM? While there is ongoing research on generalization and algorithmic capabilities of LLMs and even some hope to solve it just by scaling up a practical approach (less interesting for AI researchers) could be to give "actions" (e.g. special tokens) to the LLMs to use existing math tools for computations.
